texdoc/__init__.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+
+"""texdoc extension
+
+This is an extension to search for LaTeX documentation.
+
+Synopsis: <trigger> <query>"""
+
+import os
+import re
+import subprocess
+
+from shutil import which
+from albertv0 import *
+
+__iid__          = "PythonInterface/v0.3"
+__prettyname__   = "TeXdoc"
+__version__      = "1.0"
+__trigger__      = "td"
+__author__       = "Florian Adamsky"
+__dependencies__ = ["texdoc"]
+
+iconPath   = os.path.dirname(__file__)+"/texdoc-logo.svg"
+texdoc_cmd = ["texdoc", "-I", "-q", "-s", "-M"]
+
+if which("texdoc") is None:
+    raise Exception("'texdoc' is not in $PATH.")
+
+def handleQuery(query):
+    if not query.isTriggered:
+        return
+
+    query.disableSort()
+
+    stripped_query = query.string.strip()
+
+    if stripped_query:
+        process = subprocess.run(["texdoc", "-I", "-q", "-s", "-M", stripped_query],
+                              stdout=subprocess.PIPE)
+        texdoc_output = process.stdout.decode("utf-8")
+
+        results = []
+        for line in texdoc_output.split("\n"):
+
+            match = re.search('\t(/.*/)([\w\.-]+)\t\t', line, re.IGNORECASE)
+            if match:
+                directory = match.group(1).strip()
```

Comment:
Could there be  a case where < 2 groups are in the regex? If so, is it worth handling it before calling `.group(...)`

Reply:
I can't think of any case where the groups < 2. This matches a file and separates the directory and the filename. If texdoc has found a documentation it is always a full path with filename.